resolve_test.go

```diff
@@ -236,22 +232,33 @@ func TestResolutionHostnameWithNoWellKnownNorSRV(t *testing.T) {
 func setupFakeDNS(answerSRV bool) (cleanup func()) {
 	defaultResolver := net.DefaultResolver
 
-	// Start a DNS server with our custom handler.
-	srv := &dns.Server{Addr: fmt.Sprintf("127.0.0.1:%d", dnsPort), Net: "udp"}
-	srv.Handler = &dnsHandler{answerSRV: answerSRV}
+	// Start a DNS server with our custom handler on a random available port.
+	listenAddr, err := net.ResolveUDPAddr("udp", "127.0.0.1:0")
+	if err != nil {
+		panic(fmt.Sprintf("failed to ResolveUDPAddr: %v", err))
+	}
+	udpConn, err := net.ListenUDP("udp", listenAddr)
+	if err != nil {
+		panic(fmt.Sprintf("failed to ListenUDP: %v", err))
+	}
+
+	handler := dnsHandler{answerSRV: answerSRV}
+	srv := &dns.Server{PacketConn: udpConn, Handler: &handler}
+
 	go func() {
-		err := srv.ListenAndServe()
+		err := srv.ActivateAndServe()
 		if err != nil {
 			panic(err)
 		}
 	}()
 
 	// Redefine the default resolver so it uses our local server.
+	actualListenAddr := udpConn.LocalAddr().String()
```

Comment:
A comment for why this is necessary would be nice.

Reply:
Does it look better now?

Reply:
Yep!